p2p/discover/table.go

```diff
@@ -511,6 +516,25 @@ func (tab *Table) del(node *Node) {
 	}
 }
 
+func (b *bucket) replace(n *Node, last *Node) bool {
+	// Don't add if b already contains n.
+	if b.bump(n) {
+		return false
+	}
```

Comment:
This bump here has a tiny side effect that the node `n` will become the most recent node, even if something newer already appeared in the bucket.

However, since this check was (as far as I understand) added to prevent a double insert race, I don't think it has any serious practical implications, though I thought I'd mention it just to make sure it works as intended.